stdlib/internal/SwiftSyntax/SyntaxChildren.swift

```diff
@@ -0,0 +1,28 @@
+//===------------- SyntaxChildren.swift - Syntax Child Iterator -----------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+/// A sequence that allows iteration of the children of a Syntax node.
```

Comment:
Not a random access collection would be something like a Stack or a Queue, where you can only work on a specific region of the Data Structure. Contrasting with an Array or a List in that regard.